Makefile

```diff
@@ -228,7 +233,7 @@ test-style: lint lint-charts shellcheck
 .PHONY: lint
 lint: $(GOLANGCI_LINT)
 	# Setting timeout to 5m as default is 1m
-	$(GOLANGCI_LINT) run --timeout=5m -v
+	$(GOLANGCI_LINT) run --build-tags e2e --timeout=5m -v
```

Comment:
`golangci-lint` will not work with nested modules, so we need to have another step to run `golangci-lint` inside the `test/e2eprovider` directory.

```bash
	cd test/e2eprovider && $(GOLANGCI_LINT) run --timeout=5m -v
```

thanks to @tam7t for the pointer

Reply:
added